File: shared/python/shared/schemas/modules.py
from typing import Optional

from pydantic import BaseModel


class ModuleServiceSchema(BaseModel):
    id: int
    service_name: str
    version: str
    admin_topic: str
    ttl_seconds: int
    status: str
    created_at: Optional[str] = None
    updated_at: Optional[str] = None


__all__ = ["ModuleServiceSchema"]

File: apps/web/src/widgets/finder/api/modules.ts
import apiClient from './config'

export interface ServiceInfo {
  id: number
  service_name: string
  version: string
  admin_topic: string
  ttl_seconds: number
  status: string
  created_at: string
  updated_at: string
}

export const modulesApi = {
  listServices: async (): Promise<ServiceInfo[]> => {
    const response = await apiClient.get<ServiceInfo[]>('/modules/services')
    return response.data
  },

  getServiceDetails: async (serviceName: string): Promise<ServiceInfo> => {
    const response = await apiClient.get<ServiceInfo>(`/modules/services/${serviceName}`)
    return response.data
  },
}

